File: color.go
package main

import "fmt"

type Color = Vec3

func writeColor(pixelColor Color) {
	r := pixelColor.x()
	g := pixelColor.y()
	b := pixelColor.z()

	//translating the [0,1] component values to the byte range [0,255].

	rbyte := int(255.999 * r)
	gbyte := int(255.999 * g)
	bbyte := int(255.999 * b)

	//write out pixel color components
	fmt.Printf("%d %d %d\n", rbyte, gbyte, bbyte)
}

File: src/color.rs
use crate::vec3::Vec3;

pub type Color = Vec3;

pub fn write_color(pixel_color: &Color) {
    let r = pixel_color.x();
    let g = pixel_color.y();
    let b = pixel_color.z();

    //translating the [0,1] component values to the byte range [0,255]

    let rbyte = (255.999 * r) as i32;
    let gbyte = (255.999 * g) as i32;
    let bbyte = (255.999 * b) as i32;

    print!("{rbyte} {gbyte} {bbyte}\n");
}
